pkg/utils/yaml_utils.go

```diff
@@ -75,26 +82,99 @@ func ConvertToYAML(data any) (string, error) {
 	return string(y), nil
 }
 
-func processCustomTags(node *yaml.Node) error {
+func processCustomTags(atmosConfig *schema.AtmosConfiguration, node *yaml.Node, file string) error {
 	if node.Kind == yaml.DocumentNode && len(node.Content) > 0 {
-		return processCustomTags(node.Content[0])
+		return processCustomTags(atmosConfig, node.Content[0], file)
 	}
 
 	for i := 0; i < len(node.Content); i++ {
 		n := node.Content[i]
 
 		if SliceContainsString(AtmosYamlTags, n.Tag) {
-			n.Value = n.Tag + " " + n.Value
+			val, err := getValueWithTag(atmosConfig, n, file)
+			if err != nil {
+				return err
+			}
+			n.Value = val
 		}
 
-		if err := processCustomTags(n); err != nil {
+		if err := processCustomTags(atmosConfig, n, file); err != nil {
 			return err
 		}
 	}
 	return nil
 }
 
+func getNodeValue(tag string, f string, q string) string {
+	if q == "" {
+		return strings.TrimSpace(tag + " \"" + f + "\"")
+	}
+	return strings.TrimSpace(tag + " \"" + f + "\" \" " + q + "\"")
+}
+
+func getValueWithTag(atmosConfig *schema.AtmosConfiguration, n *yaml.Node, file string) (string, error) {
+	tag := n.Tag
+	val := n.Value
+
+	if tag == AtmosYamlFuncInclude {
+		var f string
+		q := ""
+
+		parts, err := SplitStringByDelimiter(val, ' ')
+		if err != nil {
+			return "", err
+		}
+
+		partsLen := len(parts)
+
+		if partsLen == 2 {
+			f = strings.TrimSpace(parts[0])
+			q = strings.TrimSpace(parts[1])
+		} else if partsLen == 1 {
+			f = strings.TrimSpace(parts[0])
+		} else {
+			err := fmt.Errorf("invalid number of arguments in the Atmos YAML function: %s %s. The function accepts 1 or 2 arguments", tag, val)
+			return "", err
+		}
+
+		// If absolute path is provided, check if the file exists
+		if filepath.IsAbs(f) {
+			if !FileExists(f) {
+				return "", fmt.Errorf("the function '%s %s' points to a file that does not exist", tag, val)
+			}
+			return getNodeValue(tag, f, q), nil
+		}
+
+		// Detect relative paths (relative to the manifest file) and convert to absolute paths
```

Comment:
We should move this to a reusable function. @milldr has a different implementation. Also I believe the determination differs by OS. Please see what Dan did.